test/index.ts

```diff
@@ -1774,6 +1776,8 @@ describe('Datastore', () => {
           (config.reqOpts!.mutations![0].upsert! as Entity).excludeFromIndexes,
           excludeFromIndexes
         );
+        // Reassign this to what it was so that other tests don't fail because of the mock.
+        fakeEntity.entityToEntityProto = ENTITY_TO_ENTITY_PROTO;
```

Comment:
maybe surround this test with a try/finally block to ensure this resets even if the test fails?

Reply:
Done

Reply:
Reverted. Others had comments so may chime in a bit later.

Reply:
I don't think `try`/`finally` would be a good fit here because it's sync and the test is async (that is why might reset the prop before it is actually used).

I am looked at this super closely but there are several things to explore to reset the prop IMO:
- use an `async` test fn if the test framework and the code flow support it,
- use `afterTest` + global objet to restore explicitly saved props (that would look a little hacky)

Reply:
By the last comment do you mean wrapping this test in a describe block and then adding an after hook in the describe block?

Reply:
Actually I don't think my first option would work.

Maybe something to explore:

```js
const fakeEntityInit: any = {
 // previous fakeEntity
};

const fakeEntity: any = {};

// ...

  beforeEach(() => {
    // init fakeEntity
    fakeEntity[KEY_SYMBOL] = fakeEntityInit[KEY_SYMBOL];
    for (const [key, value] of Object.entries(fakeEntityInit)) {
      fakeEntity[key] = value;
    }

    // existing code   
    createInsecureOverride = null;
    googleAuthOverride = null;
    datastore = new Datastore({
      projectId: PROJECT_ID,
      namespace: NAMESPACE,
    });
  });
```

This would ensure the `fakeEntity` is init from a clean value before each test.

Reply:
This would work. I guess we don't need to care too much about the extra run time this would add since these are tests.

Reply:
I adopted the same idea, but did it a slightly different way. We do not want to have to iterate through each property like this. We can just use `Object.assign` and that will do the trick.